cmake/add_enclave.cmake

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# Helper function to sign an enclave binary.
+#
+# Usage:
+#
+#	add_enclave(<TARGET target> <SOURCES sources> <CONFIG config>)
+#
+# The target must already exist and should be a shared object (an
+# enclave).
+#
+# Given <target> and <signconffile>, this function adds custom
+# commands to generate a signing key and call `oesign` to sign the
+# target, resulting in `<target>.signed.so`. It also adds
+# `<target>_signed` as an imported target so that it can be referenced
+# later in the CMake graph.
+# TODO: (1) Replace the name guessing logic.
+# TODO: (2) Setup the dependency using `${BIN}_signed` instead of the
+# default custom target.
+function(add_enclave)
+
+   set(options)
+   set(oneValueArgs TARGET CONFIG KEY)
+   set(multiValueArgs SOURCES)
+   cmake_parse_arguments(ENCLAVE "${options}" "${oneValueArgs}"
+                          "${multiValueArgs}" ${ARGN})
+   add_executable(${ENCLAVE_TARGET} ${ENCLAVE_SOURCES})
+   
+   if(NOT ENCLAVE_CONFIG)
```

Comment:
This is to make it so that `add_enclave` can be used with enclaves that are not signed, correct? I think so... which means we need a comment to that affect, and we should start using this for the unsigned enclaves later in the review.

Reply:
I added a comment